### DISCUSSION TITLE
Restart/abort study button

### DISCUSSION THREAD

**Discussion by marco2gandolfo** [jspsych/jsPsych#1970] [Q&A] [answered]
Hello Everybody :)

I am currently designing a study in which I would need to have in every screen a restart/abort study button. Ideally somewhere on screen there would be a button to refresh the page and restart the experiment.

I saw from the topic on the following link that Jatos.js has a function for this (' [https://forum.cogsci.nl/discussion/6194/jspsych-study-with-abort-button] (url)' ). However, since I am not using it as server I was wondering whether there was a way to do something similar without having to use Jatos. 

For example, I have been trying (probably using a wrong logic) to add a button to refresh the page within the event-related callback function : on_trial_start. [https://stackoverflow.com/questions/29884654/button-that-refreshes-the-page-on-click](url)

So far I did not manage to make it work, does someone has a hint or knows whether this is possible to implement?

Thanks a lot
Marco

**Comment by jodeleeuw** [comment, marked as answer]
Hi @marco2gandolfo,

One thing you could do is use the `display_element` parameter in `jsPsych.init({...})` to have jsPsych run the experiment in a DOM element on the page, and then you could have a button on the page that wouldn't be part of the experiment.

```html
<body>
	<div id="jspsych-target"></div>
	<button id="restart-btn">Restart the Experiment</button>
</body>
<script>
	jsPsych.init({
		display_element: 'jspsych-target'
	})
</script>
```

As far as implementing the button itself, I think that depends on what exactly you want the restart to mean. Is it just literally reloading the page, or do you want to do something else?

**Reply by marco2gandolfo** [reply]
Dear @jodeleeuw 

Thanks for your reply, that sounds exactly what I am after.
Yes my restart button could be a simple refresh, which I think could be achieved with the onclick argument?
I suppose that the display element in JSPsychinit will display the button in the screen position I say it has to be within the inline html of the button, right?

Thanks a lot for your help, as usual,

Cheers,
Marco
```

<body>
	<div id="jspsych-target"></div>
	<button id="restart-btn"; onClick= "window.location.reload();">Restart the Experiment</button>
</body>
<script>
	jsPsych.init({
		display_element: 'jspsych-target'
	})
</script>
```

**Reply by jodeleeuw** [reply]
You can control the position of the experiment and the button by modifying the style of the `<body>`, `<div>` that contains the experiment, and `<button>` elements. Is that what you are asking?

**Reply by marco2gandolfo** [reply]
Hi @jodeleeuw ,

Sorry for the late reply! this was very helpful and it works - yes it is what I was asking and was pretty trivial indeed!

Cheers,
Marco
